.github/workflows/elm_tests.yml

```diff
@@ -8,37 +8,68 @@ jobs:
 
     steps:
       - name: Checkout code
-        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
-      
-      - name: Setup Elm
-        uses: jorelali/setup-elm@cc56a4b37350f5a7c70e8cb67f64019704ca118c
+        uses: actions/checkout@e2f20e631ae6d7dd3b768f56a5d2af784dd54791
+
+      # Re-use node_modules between runs until package.json or package-lock.json changes.
+      - name: Cache node_modules
+        id: cache-node_modules
+        uses: actions/cache@9b0c1fce7a93df8e3bb8926b0d6e9d89e92f20a7
+        with:
+          path: node_modules
+          key: node_modules-${{ hashFiles('package.json', 'package-lock.json') }}
```

Comment:
When using the cache action, you usually want to setup a `restore-keys` field in addition to the `key` field like this for example:
```
restore-keys: |
  node_modules-${{ hashFiles('package.json') }}
  node_modules
```
This means that when the hash of `key` is not matched, instead of starting from scratch, it starts from one of the restore keys if matched (and `node_modules` is always matched, if present). So basically just like on you computer, you don't start from scratch. Of course, if we wait too long (I believe 30 days but not sure) even the cache associated with key `node_modules` will be gone.

Same applies for the `Cache ~/.elm` action.

Reply:
Thank you, I had never used `restore-keys`. Added.